src/catalog_api.cpp

```diff
@@ -71,35 +70,150 @@ static IRCAPIColumnDefinition ParseColumnDefinition(yyjson_val *column_def) {
+		}
+		options["use_ssl"] = Value(!path_style);
+		if (path_style) {
+			options["url_style"] = "path";
+		}
+	}
+
+	auto endpoint_it = options.find("endpoint");
+	if (endpoint_it == options.end()) {
+		return;
+	}
+	auto endpoint = endpoint_it->second.ToString();
+	if (StringUtil::StartsWith(endpoint, "http://")) {
+		endpoint = endpoint.substr(7, std::string::npos);
+	}
+	if (StringUtil::EndsWith(endpoint, "/")) {
+		endpoint = endpoint.substr(0, endpoint.size() - 1);
+	}
+	endpoint_it->second = endpoint;
+}
+
 IRCAPITableCredentials IRCAPI::GetTableCredentials(ClientContext &context, IRCatalog &catalog, const string &schema,
-                                                   const string &table, IRCCredentials credentials) {
+                                                   const string &table, const string &secret_base_name) {
 	IRCAPITableCredentials result;
 	string api_result = GetTableMetadataCached(context, catalog, schema, table, catalog.secret_name);
 	std::unique_ptr<yyjson_doc, YyjsonDocDeleter> doc(ICUtils::api_result_to_doc(api_result));
 	auto *root = yyjson_doc_get_root(doc.get());
-	auto *warehouse_credentials = yyjson_obj_get(root, "config");
-	auto credential_size = yyjson_obj_size(warehouse_credentials);
 	auto catalog_credentials = IRCatalog::GetSecret(context, catalog.secret_name);
-	if (warehouse_credentials && credential_size > 0) {
-		result.key_id = IcebergUtils::TryGetStrFromObject(warehouse_credentials, "s3.access-key-id", false);
-		result.secret = IcebergUtils::TryGetStrFromObject(warehouse_credentials, "s3.secret-access-key", false);
-		result.session_token = IcebergUtils::TryGetStrFromObject(warehouse_credentials, "s3.session-token", false);
-		if (catalog_credentials) {
-			auto kv_secret = dynamic_cast<const KeyValueSecret &>(*catalog_credentials->secret);
-			auto region = kv_secret.TryGetValue("region").ToString();
-			result.region = region;
+
+	// Mapping from config key to a duckdb secret option
+
+	case_insensitive_map_t<Value> config_options;
+	auto *config_val = yyjson_obj_get(root, "config");
+	if (config_val && catalog_credentials) {
+		auto kv_secret = dynamic_cast<const KeyValueSecret &>(*catalog_credentials->secret);
+		auto region = kv_secret.TryGetValue("region").ToString();
+		config_options["region"] = region;
+	}
+	ParseConfigOptions(config_val, config_options);
+
+	auto *storage_credentials = yyjson_obj_get(root, "storage-credentials");
+	auto storage_credentials_size = yyjson_arr_size(storage_credentials);
+	if (storage_credentials && storage_credentials_size > 0) {
+		yyjson_val *storage_credential;
+		size_t index, max;
+		yyjson_arr_foreach(storage_credentials, index, max, storage_credential) {
+			auto *sc_prefix = yyjson_obj_get(storage_credential, "prefix");
+			if (!sc_prefix) {
+				throw InternalException("required property 'prefix' is missing from the StorageCredential schema");
```

Comment:
Should this be an InternalException? I feel like since this is a catalog error, we shouldn't invalidate the whole database

Reply:
I'd go for IOException, but I don't know what else fits, because this means the REST catalog is broken, that's a required property

Reply:
went with InvalidInput